packages/block-editor/src/hooks/behaviors.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { SelectControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { InspectorControls } from '../components';
+import { store as blockEditorStore } from '../store';
+
+/**
+ * Override the default edit UI to include a new block inspector control for
+ * assigning behaviors to blocks if behaviors are enabled in the theme.json.
+ *
+ * Currently, only the `core/image` block is supported.
+ *
+ * @param {WPComponent} BlockEdit Original component.
+ *
+ * @return {WPComponent} Wrapped component.
+ */
+export const withBehaviors = createHigherOrderComponent( ( BlockEdit ) => {
+	return ( props ) => {
+		// Only add behaviors to the core/image block.
+		if ( props.name !== 'core/image' ) {
+			return <BlockEdit { ...props } />;
+		}
+
+		// Check the value of `settings.blocks.core/image.behaviorsUIEnabled` in the
+		// theme.json. If false, do not add the behaviors inspector control.
+		const settings = select( blockEditorStore ).getSettings();
+		if (
+			! settings?.__experimentalFeatures?.blocks?.[ props.name ]
+				?.behaviorsUIEnabled
+		) {
+			return <BlockEdit { ...props } />;
+		}
+
+		const { behaviors: blockBehaviors } = props.attributes;
+
+		// Get the theme behaviors from the theme.json.
+		const themeBehaviors = select( blockEditorStore ).getBehaviors();
+
+		// By default, use the block behaviors.
+		// If the theme has behaviors, but the block does not, use the theme behaviors.
+		const behaviors = blockBehaviors || themeBehaviors || {};
+
+		return (
+			<>
+				<BlockEdit { ...props } />
+				<InspectorControls group="advanced">
+					<SelectControl
+						__nextHasNoMarginBottom
+						label={ __( 'Behaviors' ) }
+						// At the moment we are only supporting one behavior (lightbox)
+						value={ behaviors?.lightbox ? 'LIGHTBOX' : '' }
+						options={ Object.keys( behaviors )
+							.map( ( behavior ) => ( {
+								value: behavior,
+								label: behavior.toUpperCase(),
+							} ) )
+							.concat( {
+								value: '',
+								label: __( 'No behaviors' ),
```

Comment:
Why are the behavior names uppercase in the UI? Is it needed? Maybe we can define their UI name somewhere.

Reply:
There's no specific reason other than that I thought it looked better 🙁. Perhaps they could just be capitalized like the names of the specific image sizes:

<img width="269" alt="Screenshot 2023-05-22 at 15 48 51" src="https://github.com/WordPress/gutenberg/assets/5417266/d2756c00-0cb1-4f41-8933-c9095a04c8b7">


<img width="265" alt="Screenshot 2023-05-22 at 15 46 00" src="https://github.com/WordPress/gutenberg/assets/5417266/5854babe-2fde-4cf4-a8e5-3ad657eea1bc">

Reply:
OK, I've capitalized the behavior names now. I think it's more consistent this way.